gtsam/hybrid/tests/testHybridFactorGraph.cpp

```diff
@@ -335,10 +335,36 @@ TEST(DCGaussianElimination, Eliminate_fully) {
   // EXPECT_LONGS_EQUAL(1, result.second->size());
 }
 
+/* ****************************************************************************/
+/// Test the toDecisionTreeFactor method
+TEST(HybridFactorGraph, ToDecisionTreeFactor) {
+  Switching self(3);
+
+  // Clear out discrete factors since sum() cannot hanldle those
+  HybridFactorGraph linearizedFactorGraph(
+      NonlinearFactorGraph(), DiscreteFactorGraph(),
+      self.linearizedFactorGraph.dcGraph(),
+      self.linearizedFactorGraph.gaussianGraph());
+
+  auto decisionTreeFactor = linearizedFactorGraph.toDecisionTreeFactor();
+
+  auto allAssignments = cartesianProduct(linearizedFactorGraph.discreteKeys());
+
+  // Regressions: test all 4 assignments.
```

Comment:
So, a regression test is not what is needed here, but something like:
- create nonlinear factor graph for m1=0, m2=1
- linearize, optimize, get true error
- compare with error stored in decision tree

if this works for 01, we can trust it to work for all 4 combinations.

Reply:
I was thinking of doing this but the errors are currently too close to each other to really make that distinction.

I'll update the noise models so we get error values in the 1e-2 range that can be more easily compared, after that adding this separate graph should make a lot of sense.

Reply:
I'm (reluctantly) fine with it as is, just for future reference, a *real* test like that is always preferred to hardcoding values; how do you *know* those values are correct?

Reply:
Yes I completely agree. This test is the equivalent of a strawman argument and wanted to use this time to brainstorm a proper test. I got an idea while I was out running. :)

Reply:
Something like

- create nonlinear factor graph for m1=0, m2=1
- linearize, optimize, get true error
- compare with error stored in decision tree
perhaps ? :-)

Reply:
All done! You should like this new test a lot more 😄